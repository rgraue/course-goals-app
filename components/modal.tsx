import { Modal, View, Button, Text } from "react-native";

export const GoalModal = ({isVisible, goal, id, closeModalHandler, deleteGoalHandler}) => {

  return (
    <View>
      <Modal visible={isVisible}>
        <Text>{goal}</Text>
        <Button title="Close" onPress={closeModalHandler.bind(this)}/>
        <Button title="Goal Completed" onPress={deleteGoalHandler.bind(this, id)}/>
      </Modal>
    </View>
  )
}