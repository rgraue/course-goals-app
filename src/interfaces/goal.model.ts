export interface Goal {
  id: string;
  name: string;
}
