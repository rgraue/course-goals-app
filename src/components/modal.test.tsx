import React from 'react';
import { ListRenderItemInfo } from 'react-native';
import { render, screen, fireEvent } from '@testing-library/react-native';
import { GoalModal } from './modal';
import { Goal } from '../interfaces';

const mockCloseFn = jest.fn();
const mockDeleteFn = jest.fn();

const mockGoal: Partial<ListRenderItemInfo<Goal>> = {
  item: {
    name: 'test goal name',
    id: '2',
  },
};

test('Modal Component', async () => {
  const tree = render(
    <GoalModal
      isVisible={true}
      goal={mockGoal.item?.name}
      id={mockGoal.item?.id}
      closeModalHandler={mockCloseFn}
      deleteGoalHandler={mockDeleteFn}
    />,
  );

  const closeButton = await screen.findByLabelText('Close Btn');
  const deleteButton = await screen.findByLabelText('Delete Btn');

  fireEvent.press(closeButton);
  fireEvent.press(deleteButton);

  expect(mockCloseFn).toBeCalledTimes(1);
  expect(mockDeleteFn).toBeCalledTimes(1);
  expect(mockDeleteFn).toBeCalledWith('2');
  expect(tree).toMatchSnapshot();
});
