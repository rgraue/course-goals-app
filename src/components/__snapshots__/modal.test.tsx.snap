// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D9C3D9",
        "borderRadius": 5,
        "flexDirection": "column",
        "height": "12%",
        "justifyContent": "center",
        "margin": "5%",
        "marginTop": "70%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "grey",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": "3%",
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        test goal name
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "marginLeft": "5%",
            "marginRight": "5%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Close
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginLeft": "5%",
            "marginRight": "5%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Goal Completed
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
